### ISSUE TITLE
Incorrect model ENUM validation

### ISSUE THREAD

**Issue by dercoder** [phalcon/cphalcon#11505]
There is an issue with ENUM values handling.
This scenario wont work:

Phalcon Version: ```2.0.10```

Database:
```
CREATE TABLE `robots` (
	`id` INT(11) NOT NULL AUTO_INCREMENT,
	`name` VARCHAR(50) NULL DEFAULT NULL,
	`type` ENUM('internet','member','prepaid','terminal') NOT NULL,
	PRIMARY KEY (`id`)
)
```

Model:
```
class Robots extends Phalcon\Mvc\Model
{
    public $id;
    public $name;
    public $type;
}
```

Example:
```
$robot = new Robots;
$robot->name = 'TestMember';
$robot->type = 'member';

if(!$robot->save())
{
     echo $robot->getMessages();
}
```
Result:
```
type is required
```

When i remove "internet" from ENUM types in database, the code is working fine.
I think its because the string "INT" exists in word "INTernet" like in this old issue: https://github.com/phalcon/cphalcon/issues/709

**Comment by dercoder** [comment]
Second Test:

$robot = new Robots();
$metaData = $robot->getModelsMetaData();
$dataTypes = $metaData->getDataTypes($robot);
```
var_dump($dataTypes['id'] == \Phalcon\Db\Column::TYPE_INTEGER);
var_dump($dataTypes['name'] == \Phalcon\Db\Column::TYPE_VARCHAR);
var_dump($dataTypes['type'] == \Phalcon\Db\Column::TYPE_CHAR);
var_dump($dataTypes['type'] == \Phalcon\Db\Column::TYPE_INTEGER);
```

Result with ENUM ('member','prepaid','terminal','internet')
```
bool(true)
bool(true)
bool(false)
bool(true)
```

Result with ENUM ('member','prepaid','terminal')
```
bool(true)
bool(true)
bool(true)
bool(false)
```

**Comment by dercoder** [comment]
I think i have located the "bug":
https://github.com/phalcon/cphalcon/blob/master/phalcon/db/adapter/pdo/mysql.zep#L136

**Comment by kristofferingemansson** [comment]
Ran in to the same bug with a column type of ENUM('int', 'decimal', 'datetime', 'string', 'text').

Detection of enum column type needs to happen before detection of the other types.

**Comment by dercoder** [comment]
Yes or use some regex to detect to correct type.

**Comment by sergeyklay** [comment]
Fixed in `2.1.x` and `2.0.x` branches
